### ISSUE TITLE
SDL: HiDPI support on Wayland (blurry fonts)

### ISSUE THREAD

**Issue by hch12907** [hrydgard/ppsspp#17650]
### Game or games this happens in

None

### What area of the game / PPSSPP

When I am using PPSSPP on Wayland with HiDPI scaling, there is a general blurriness on everything in the main menu,.

I have a hacky build (with hardcoded DPI and stuff...) that showcases how it should look like compared to the latest stable build:

![image](https://github.com/hrydgard/ppsspp/assets/7030150/8aff35e5-3a6a-4b60-b9d9-3a43099c8074)

![image](https://github.com/hrydgard/ppsspp/assets/7030150/12744f61-a2ab-4d8c-975a-83aaf846a90d)

### What should happen

The text should look sharp.

### Platform

Linux / BSD

### Mobile phone model or graphics card

It's irrelevant, but AMD Renoir iGPU

### PPSSPP version affected

42d4b5d41d7

### Graphics backend (3D API)

Other

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [x] Try resetting settings or older versions and include if the issue is related.
- [x] Try without any cheats and without loading any save states.
- [x] Include logs or screenshots of issue.

**Comment by hch12907** [comment]
I have a commit for this in ba3203d.

For some reason the compiler chokes on `SDL_Vulkan_GetDrawableSize()` in SDLMain.cpp though...

**Comment by hrydgard** [comment]
Ah, hm. Do feel free to make a pull request for it anyway, and let's use CI to figure out what platforms have trouble..

**Comment by hch12907** [comment]
I closed this because HiDPI support on Wayland was added in #17651. Ingame graphics look crisp now as expected, but the text in main menu still looked somewhat soft:

![image](https://github.com/hrydgard/ppsspp/assets/7030150/e8f4dcde-ff67-4c47-9618-c4343506cf3b)

Interestingly, the borders of each UI element (buttons etc) look pretty sharp. The only thing that remains to be fixed on HiDPI is text rendering.
